sled-agent/src/bin/sled-agent.rs

```diff
@@ -54,20 +54,29 @@ enum Args {
     },
     /// Runs the Sled Agent server.
     Run {
+        /// UUID of the Sled Agent.
         #[structopt(name = "SA_UUID", parse(try_from_str))]
         uuid: Uuid,
 
+        /// Socket address of the bootstrap agent.
         #[structopt(name = "BA_IP:PORT", parse(try_from_str))]
         bootstrap_agent_addr: SocketAddr,
 
+        /// Socket address of the sled agent.
         #[structopt(name = "SA_IP:PORT", parse(try_from_str))]
         sled_agent_addr: SocketAddr,
 
+        /// Socket address of Nexus.
         #[structopt(name = "NEXUS_IP:PORT", parse(try_from_str))]
         nexus_addr: SocketAddr,
 
+        /// Optional VLAN, tagged on all guest NICs.
         #[structopt(long = "vlan")]
         vlan: Option<VlanID>,
+
+        /// Optional list of zpools managed by Sled agent.
+        #[structopt(long = "zpools", name = "zpools", parse(try_from_str))]
+        zpools: Option<Vec<String>>,
     },
```

Comment:
Does the sled agent need to know/advertise the IP/Port for the storage agent?

Reply:
Yes, it does: see line 330ish of the storage_manager. We allocate IPs for each of the datasets, and report them to Nexus.